### PR TITLE
Set up CI and deploy

### PR DESCRIPTION
**Pull request by bhoggard**
This won't deploy until we merge into master. If it looks OK, I will work on making sure it works when that happens on TravisCI.

### CODE REVIEW

**Review by acjay — APPROVED** [review]
Nice! Can't wait to get that automatic publishing going. Does it work automatically for snapshots? Because that would be amazing.

**Comment by bhoggard** [comment]
The docs say

> Note: If your project version has "SNAPSHOT" suffix, your project will be published to the snapshot repository of Sonatype, and you cannot use sonatypeRelease command.

So, should I set it up to do that on a successful merge of the development branch?

**Comment by acjay** [comment]
Yeah, that would be great!

File: ci/before-install.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [ "$TRAVIS_BRANCH" = 'master' ] && [ "$TRAVIS_PULL_REQUEST" == 'false' ]; then
+  openssl aes-256-cbc -K $encrypted_c25aa352ba20_key -iv $encrypted_c25aa352ba20_iv
```
Comment (izakp) [inline ci/before-install.sh:6]: I take it `$encrypted_c25aa352ba20_key` / `$encrypted_c25aa352ba20_iv` are injected into the build environment somehow?

**Review by izakp — COMMENTED** [review]
Great work!

File: ci/before-install.sh (hunk 1 above)
Comment (bhoggard) [reply ci/before-install.sh:6]: Travis creates those env vars when you use their file encryption.

**Pull request merged**